packages/core-mobile/app/vmModule/ApprovalController/handleAvalancheSendTransaction.ts

```diff
@@ -0,0 +1,81 @@
+import { rpcErrors } from '@metamask/rpc-errors'
+import { Account } from 'store/account/types'
+import walletService from 'services/wallet/WalletService'
+import networkService from 'services/network/NetworkService'
+import { ApprovalResponse, Hex, RpcRequest } from '@avalabs/vm-module-types'
+import { EVM, EVMUnsignedTx, UnsignedTx } from '@avalabs/avalanchejs'
+import { Avalanche } from '@avalabs/core-wallets-sdk'
+
+export const handleAvalancheSendTransaction = async ({
+  request,
+  unsignedTxJson,
+  vm,
+  account,
+  isTestnet,
+  resolve
+}: {
+  request: RpcRequest
+  unsignedTxJson: string
+  vm: 'EVM' | 'AVM' | 'PVM'
+  account: Account
+  isTestnet?: boolean
+  resolve: (value: ApprovalResponse) => void
+}): Promise<void> => {
+  try {
+    // @ts-ignore
+    const externalIndices = request.params?.externalIndices ?? []
+    // @ts-ignore
+    const internalIndices = request.params?.internalIndices ?? []
+
+    // Parse the json into a tx object
+    const unsignedTx =
+      vm === EVM
+        ? EVMUnsignedTx.fromJSON(unsignedTxJson)
+        : UnsignedTx.fromJSON(unsignedTxJson)
+
+    const hasMultipleAddresses =
+      unsignedTx.addressMaps.getAddresses().length > 1
+
+    if (
+      hasMultipleAddresses &&
+      !externalIndices.length &&
+      !internalIndices.length
+    ) {
+      throw new Error(
+        'Transaction contains multiple addresses, but indices were not provided'
+      )
+    }
+
+    const signedTransactionJson = await walletService.sign({
+      // Must tell it is avalanche network
+      network: networkService.getAvalancheNetworkP(isTestnet ?? false),
```

Comment:
We can't use the network object here because the transaction must be signed with the Avalanche signer, even if the `avalanche_sendTransaction` is on the C-Chain (e.g., `Export from C to X`). The reason we can pass only the P-Chain network here is that this object is solely for obtaining the Avalanche signer, and both PVM and AVM use the same signer. It’s the same in the seedless wallet, which uses the same key (`cs.Secp256k1.Ava`) for signing on both X and P chains.

I agree this can be confusing. I’ve added some supplementary comments, but I’m open to better solutions if you have any.